spec/integration/tag_counts_spec.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+RSpec.describe "Updating tag counts" do
```

Comment:
Integration test for making sense of what happens when regular topics are moved into different categories or when they are converted to private messages.

Reply:
Hah I had to add one of these for my secure upload change to deal with the more complicated scenarios too :+1: